### PR TITLE
Fix/matmul

### PR DESCRIPTION
**Pull request by xhuohai**
1. Fix type infer of matmul
2. Add matmul to conv2d transforms
3. Reshape batch matmul to non-batch

### CODE REVIEW

File: src/Nncase.Evaluator/Math/MatMul.cs [hunk 1]
```diff
@@ -78,12 +78,16 @@ private IRType Visit(TensorType lhs, TensorType rhs)
             return new TensorType(lhs.DType, new[] { lhs.Shape[0], rhs.Shape[1] });
         }
 
-        var bigShape = lhs.Shape.Rank > rhs.Shape.Rank
-            ? lhs.Shape
-            : rhs.Shape;
+        var lhsShape = lhs.Shape.Rank >= rhs.Shape.Rank ? lhs.Shape.ToArray() : Enumerable.Repeat((Dimension)1, rhs.Shape.Rank - lhs.Shape.Rank).Concat(lhs.Shape).ToArray();
+        var rhsShape = lhs.Shape.Rank <= rhs.Shape.Rank ? rhs.Shape.ToArray() : Enumerable.Repeat((Dimension)1, lhs.Shape.Rank - rhs.Shape.Rank).Concat(rhs.Shape).ToArray();
+
+        var bigShape = Enumerable.Zip(lhsShape, rhsShape).ToArray()[..^2].Select(t =>
```
Comment (FusionBolt) [inline src/Nncase.Evaluator/Math/MatMul.cs:84]: `Enumerable.Zip(lhsShape, rhsShape).Take(2)` is better

**Review by FusionBolt — CHANGES_REQUESTED** [review]
Should add test for "Fix type infer of matmul"

File: src/Nncase.Evaluator/Math/MatMul.cs (hunk 1 above)
Comment (xhuohai) [reply src/Nncase.Evaluator/Math/MatMul.cs]: done

**Review by zhen8838 — DISMISSED** [review]

**Comment by xhuohai** [comment]
> Should add test for "Fix type infer of matmul"

added

**Review by FusionBolt — APPROVED** [review]

**Pull request merged**